### PR TITLE
Add `yii\web\Request::getOrigin()` method that returns HTTP_ORIGIN of current CORS request

### PR DESCRIPTION
**Pull request by yyxx9988**
>The Origin request header indicates where a fetch originates from. It doesn't include any path information, but only the server name. It is sent with CORS requests, as well as with POST requests. It is similar to the Referer header, but, unlike this header, it doesn't disclose the whole path.

From https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin

### Working code samples
```php
<?php

// We'll be granting access to only the arunranga.com domain 
// which we think is safe to access this resource as application/xml

if($_SERVER['HTTP_ORIGIN'] == "http://arunranga.com") {
    header('Access-Control-Allow-Origin: http://arunranga.com');
    header('Content-type: application/xml');
    readfile('arunerDotNetResource.xml');
} else {    
  header('Content-Type: text/html');
  echo "<html>";
  echo "<head>";
  echo "   <title>Another Resource</title>";
  echo "</head>";
  echo "<body>",
       "<p>This resource behaves two-fold:";
  echo "<ul>",
         "<li>If accessed from <code>http://arunranga.com</code> it returns an XML document</li>";
  echo   "<li>If accessed from any other origin including from simply typing in the URL into the browser's address bar,";
  echo   "you get this HTML document</li>", 
       "</ul>",
     "</body>",
   "</html>";
}
?>
```
See https://developer.mozilla.org/en-US/docs/Web/HTTP/Server-Side_Access_Control for more info.

### CODE REVIEW

**Comment by SamMousa** [comment]
This will break when / if  #13780 gets merged.
Could you use the header collection instead?

**Comment by yyxx9988** [comment]
Of course. Close this pr if it break other :）

**Comment by SamMousa** [comment]
@yyxx9988 This PR doesn't break anything, but it will be broken by the other PR.
Could you refactor it to use the header collection that is available in the request object?

**Comment by yyxx9988** [comment]
> Could you refactor it to use the header collection that is available in the request object?

@SamMousa ...I'm sorry. But did you mean this:
```php
public function getOrigin()
{
    return $this->getHeaders()->get('HTTP_REFERER');
}
```

**Comment by SamMousa** [comment]
Almost:
````
 return $this->getHeaders()->get('referer');
 return $this->getHeaders()->get('origin');

````
etc..

**Comment by yyxx9988** [comment]
@SamMousa Call me Newbie, please...😅😅😅

**Comment by samdark** [comment]
https://travis-ci.org/yiisoft/yii2/jobs/217006172

File: framework/web/Request.php [hunk 1]
```diff
@@ -905,6 +906,15 @@ public function getReferrer()
     }
 
     /**
+     * Returns the URL origin.
```
Comment (cebe) [inline framework/web/Request.php]: should explain that it is taken from CORS related header.

File: framework/web/Request.php [hunk 2]
```diff
@@ -905,6 +906,15 @@ public function getReferrer()
...
+     * @return string|null URL origin, null if not available
```
Comment (cebe) [inline framework/web/Request.php]: `@since` PHPdoc annotation is needed here.

**Review by cebe — CHANGES_REQUESTED** [review]
also a CHANGELOG entry is missing

**Comment by yyxx9988** [comment]
commits added 😀

**Review by samdark — APPROVED** [review]

**Review by cebe — APPROVED** [review]

**Comment by cebe** [comment]
Merged, thank you!

**Pull request closed without merging**